### ISSUE TITLE
Can't pass objects or function in params like in react-navigation

### ISSUE THREAD

**Issue by kations** [expo/router#304]
### Summary

I have some use cases where i open a SearchModal an want to pass a function like this:

router.push({
        pathname: "/search",
        params: {
            onSelect: ({ symbol }) => {
              setValue((v) => {
                return [...v, { symbol, amount: 1 }];
          });
       },
   },
});

to grab the selected Item in the parent route an update a form. With react-navigation this works, expo-router tries to stringify everything. Would be great to have a option to pass params the "classic" way.

### Minimal reproducible example

router.push({
        pathname: "/search",
        params: {
            onSelect: ({ symbol }) => {
              setValue((v) => {
                return [...v, { symbol, amount: 1 }];
          });
       },
   },
});

**Comment by EvanBacon** [comment]
Yes this is not supported, all params must be serializable. Non-serializable props break assumptions about deep linking, hydration, in-app linking, web, redirects, etc. Use [React Context](https://reactjs.org/docs/context.html) to pass around anything more complex.

**Comment by kations** [comment]
@EvanBacon I get the same result with some param ping pong 👍🏻 do you plan to bring the merge option to router.push?

navigation.navigate({
   name: 'Home',
   params: { post: postText },
   merge: true,
});

in my current solution I need to pass all params from the FormRoute to the SearchRoute to pass them back to the FormRoute after the user selects an item, otherwise I would lose my previous Form params

**Comment by Wenceslauu** [comment]
@kations How did you avoid the 'never' typescript errors when using .navigate()?
